sdk/core/azure-core-http-vertx/src/main/java/com/azure/core/http/vertx/implementation/VertxRequestWriteSubscriber.java

```diff
@@ -122,19 +130,50 @@ private void onErrorInternal(Throwable throwable) {
         // code 2 and greater are completion states which means the error should be dropped as we already completed.
         if (state.code >= 2) {
             Operators.onErrorDropped(throwable, Context.of(contextView));
+
+            // Also, even though Reactor may have an operator for the dropped error or will be logged by Operators
+            // itself, we should log as well as at least this will help associate the error with this class.
+            LOGGER.atError().log(() -> "VertxRequestWriteSubscriber dropped an exception as it already reached a "
+                + "completion state.", throwable);
         }
 
         this.state = State.ERROR;
         if (state != State.WRITING) {
             resetRequest(throwable);
         } else {
-            error = throwable;
+            if (error != null) {
+                // Already saw another error while writing, add this as a suppressed exception.
+                error.addSuppressed(throwable);
+            } else {
+                // First error seen while writing, maintain it for future use.
+                error = throwable;
+            }
         }
     }
 
     private void resetRequest(Throwable throwable) {
         subscription.cancel();
-        promise.fail(throwable);
+        if (!promise.tryFail(throwable)) {
+            // Seems the promise has already completed in some form.
+            // Attempt to associate this error with the existing failure.
+            Throwable cause = promise.future().cause();
+            if (cause != null) {
+                cause.addSuppressed(throwable);
+
+                // Also, even though the exception was added as a suppressed exception to the failed Promise, we should
+                // log as well as at least this will help associate the error with this class.
+                LOGGER.atError().log(() -> "VertxRequestWriteSubscriber added an exception as a suppressed exception "
+                    + "as the Promise already failed.", throwable);
+            } else {
+                // Turns out the future was completed as successfully externally, drop the error.
+                Operators.onErrorDropped(LOGGER.logThrowableAsError(throwable), Context.of(contextView));
+
+                // Also, even though Reactor may have an operator for the dropped error or will be logged by Operators
+                // itself, we should log as well as at least this will help associate the error with this class.
+                LOGGER.atError().log(() -> "VertxRequestWriteSubscriber dropped an exception as the Promise already "
```

Comment:
this sounds like a `debug` log - everything is successful, why log an error?

Reply:
Changed to info, while the `Promise` succeeded this still failed